### PR TITLE
Migrate VM Tools screens to be null safe

### PR DESCRIPTION
**Pull request by bkonyi**
Fixes https://github.com/flutter/devtools/issues/3780

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view.dart [hunk 1]
```diff
@@ -352,9 +353,9 @@ class _IsolatePortsWidgetState extends State<IsolatePortsWidget> {
   final id = _PortIDColumn();
   final name = _PortNameColumn();
 
-  final selectedPort = ValueNotifier<InstanceRef>(null);
+  final selectedPort = ValueNotifier<InstanceRef?>(null);
 
-  List<ColumnData<InstanceRef>> get columns => [
+  List<ColumnData<InstanceRef?>> get columns => [
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view.dart:358]: can InstanceRef be non-nullable

File: packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view.dart [hunk 2]
```diff
@@ -371,15 +372,15 @@ class _IsolatePortsWidgetState extends State<IsolatePortsWidget> {
               child: Split(
                 axis: Axis.horizontal,
                 children: [
-                  FlatTable<InstanceRef>(
+                  FlatTable<InstanceRef?>(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view.dart:375]: same question here and for types in callbacks below

File: packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view_controller.dart [hunk 3]
```diff
@@ -51,42 +50,42 @@ class IsolateStatisticsViewController extends DisposableController
...
     _refreshing.value = true;
 
+    final isolateId = isolateRef.id!;
     // Retrieve updated isolate information and refresh the page.
-    _isolate = await _service.getIsolate(isolateRef.id);
+    _isolate = await _service.getIsolate(isolateId);
+    final isolate = _isolate!;
     _updateTagCounters(isolate);
-    _ports = (await _service.getPorts(_isolate.id)).ports;
+    _ports = (await _service.getPorts(isolateId)).ports!;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view_controller.dart:63]: just want to verify isolateRef.id == isolate.id? before this change we were getting the id from the Isolate, and this changes behavior to grab the id from the isolateRef

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart [hunk 4]
```diff
@@ -75,14 +73,17 @@ class VMDeveloperToolsScreenBody extends StatefulWidget {
...
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     final newController = Provider.of<VMDeveloperToolsController>(context);
-    if (newController == controller) return;
+    // Don't access `controller` if we haven't already initialized it.
+    if (initialized && newController == controller) return;
     controller = newController;
+    initialized = true;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart:86]: FYI @polina-c we should use this pattern elsewhere when we are initializing `controller` in widget state classes. (https://github.com/flutter/devtools/pull/3822#discussion_r823183816)

Comment (polina-c) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart:86]: So, the idea is to set it just once, right?
Good to know thank you. Thank you!

File: packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view_controller.dart (hunk 3 above)
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view_controller.dart:63]: Yes, isolate IDs are stable.

File: packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view.dart (hunk 1 above)
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view.dart]: Yup, done.

File: packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart:86]: We will still update `controller` if the dependencies changed (i.e. if the provided controller changed), but with this pattern we will not access `controller` to check equality before it is initialized, thus allowing it to be late and non-nullable.

**Pull request merged**